src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,89 @@
-import React from 'react';
+import { Comment, CommentData } from '../types/Comment';
+import React, { Dispatch, FormEvent, useState } from 'react';
+
+import { Post } from '../types/Post';
+import { postComment } from '../api/postComment';
+
+interface Props {
+  selectedPost: Post;
+  setComments: Dispatch<React.SetStateAction<Comment[]>>;
+  setIsError: Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPost,
+  setComments,
+  setIsError,
+}) => {
+  const [formState, setFormState] = useState({
+    name: { value: '', error: '' },
+    email: { value: '', error: '' },
+    text: { value: '', error: '' },
+    isLoading: false,
+  });
+
+  const resetForm = () => {
+    setFormState({
+      name: { value: '', error: '' },
+      email: { value: '', error: '' },
+      text: { value: '', error: '' },
+      isLoading: false,
+    });
+  };
+
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const newFormState = { ...formState };
+
+    if (!newFormState.name.value) {
+      newFormState.name.error = 'Name is required';
+    }
+
+    if (!newFormState.email.value) {
+      newFormState.email.error = 'Email is required';
+    }
+
+    if (!newFormState.text.value) {
+      newFormState.text.error = 'Enter some text';
+    }
+
+    if (
+      newFormState.name.error ||
+      newFormState.email.error ||
+      newFormState.text.error
+    ) {
+      setFormState(newFormState);
+
+      return;
+    }
+
+    setFormState({ ...newFormState, isLoading: true });
+
+    const newComment: CommentData = {
+      name: newFormState.name.value,
+      email: newFormState.email.value,
+      body: newFormState.text.value,
+      postId: selectedPost.id,
+    };
+
+    try {
+      const res = await postComment('/comments', newComment);
+
+      setComments(prevComments => [...prevComments, res]);
+    } catch (error) {
+      setIsError(true);
+    } finally {
+      setFormState({ ...newFormState, isLoading: false });
+      setFormState(prevState => ({
+        ...prevState,
+        text: { ...prevState.text, value: '' },
+      }));
+    }
+  };
 
-export const NewCommentForm: React.FC = () => {
   return (
-    <form data-cy="NewCommentForm">
+    <form data-cy="NewCommentForm" onSubmit={handleSubmit}>
       <div className="field" data-cy="NameField">
```

Comment:
Optional: You can create NameField component and re-use it in your form (for now it looks you are copy-pasting)